tests/phpunit/tests/media.php

```diff
@@ -3462,6 +3457,161 @@ public function data_wp_get_attachment_image_decoding_attr() {
 		);
 	}
 
+	/**
+	 * Tests that the `wp_get_loading_optimization_attributes()` function should add
+	 * the 'decoding' attribute.
+	 *
+	 * @ticket 58892
+	 *
+	 * @dataProvider data_add_decoding_attr
+	 *
+	 * @param string $tag_name The tag name.
+	 * @param string $decoding The value for the 'decoding' attribute. 'no value' for default.
+	 * @param string $expected The expected `img` tag.
+	 * @param string $message  The test message.
+	 */
+	public function test_add_decoding_attr( $tag_name, $decoding, $expected, $message ) {
+		add_filter(
+			'wp_img_tag_add_decoding_attr',
+			static function ( $value ) use ( $decoding ) {
+				return $decoding;
+			}
+		);
```

Comment:
See my reply in https://github.com/WordPress/wordpress-develop/pull/4991#discussion_r1309477342, this filter is not the right way to test this function, as it should not be part of this function since it is historically more specific than that.

Instead, you should pass the `$decoding` value within the `$attr` array that you pass to `wp_get_loading_optimization_attributes()`. Something like:
```php
$attr = array(
	'width'  => 1,
	'height' => 1,
);
if ( $decoding !== null ) {
	$attr['decoding'] = $decoding;
}
```